src/CollapsableMixin.js

```diff
@@ -1,103 +1,133 @@
 var React = require('react');
 var TransitionEvents = require('./utils/TransitionEvents');
 
+/* This mixin keeps track of its own 'internal' expanded state.  This is to
+ * help enable running css animations on the collapsable component.  The
+ * expanded state /cannot/ bet set /before/ the mixin has had a chance to
+ * ensure that dimensions have been set on the collaspable component.  This
+ * is due to how browers interact with css transitions.  If the mixin does 
+ * not share an 'expanded' state then it makes it a lot easier to handle the
+ * order in which the dimension and `expanded` information is set.
+
+ * This pattern requires the owner of this mixin to manually call the
+ * internalToggle() function to toggle the expanded/collapsed state instead
+ * of relying upon props propigation.
+ *
+ * It is suggested that the component which uses this mixin keep its own
+ * expanded state (which could be from props or state) and call the
+ * internalToggle() function in componentDidUpdate().  Ex:
+
+   componentDidUpdate: function(prevProps, prevState){
+     var wasExpanded = prevState.expanded;
+     var isExpanded = this.state.expanded;
+     if(wasExpanded != isExpanded){
+       this.internalToggle();
+     }
+   }
+*/
```

Comment:
>I can't say that I completely understand why we can't continue to allow expanded to be used on the components that use this mixin.

They can use their own `expanded`.  The difference is previously this mixin was relying on that `external expanded` property instead of its own.

I know this feels a little hacky, but if you look at the [bootstrap collapse](https://github.com/twbs/bootstrap/blob/master/js/collapse.js) component this is exactly how they do it.  It is just a bit less code since they're not working within a structured component framework and not keeping track of state.

If the component doesn't keep its own representation of expanded then there is no way to set `node.style['height'] = '150px'` *before* the `collapsing=true` state is set.  If that fixed height style isn't set *before* the `collapsing=true` state is set, Safari will *never* run the css animation.  The css transition is defined on the `collapsing` style.

Trying to set `node.style['height'] = '150px'` in `componentWillUpdate` doesn't work, I believe because the `collapsing` class and the dimension is set essentially at the same time.  Safari just won't run the animation.  Chrome, and I think Firefox, handle this fine just not Safari.

This wasn't an issue previously since the dimension was not being removed.  Clearing out the dimension with `node.style['height'] = ''` makes all this unfortunately necessary (at least this is the only way I could get it working).

We can change the internal expanded to so some crazy name, no problem.